### PR TITLE
Feat/https proxy

### PR DESCRIPTION
**Pull request by sashokbg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,4 +75,13 @@ download-tgz search tgz
...
\ No newline at end of file
+[downloads-url]: https://npmjs.org/package/node-tgz-downloader
+
+### Proxy Configuration
+
+The tool supports the **http_proxy** environment variable.
+Do not forget to specify protocol and port.
+
+```bash
+export https_proxy='http://proxy_url:port'
```
Comment (Meir017) [inline README.md:86]: this example should probably use `https` for the protocol to be more clear

File: lib/download-file.js [hunk 2]
```diff
@@ -39,7 +40,12 @@ function downloadFileAsync(file, options = {}) {
       new Promise(x => fileClose = x),
       new Promise(x => responseEnd = x),
     ];
-    const request = req.get(file, (response) => {
+
+    let reqOptions = { agent: proxy.proxyEnabled() ? proxy.proxyAgent : req.globalAgent };
```
Comment (Meir017) [inline lib/download-file.js:44]: let's move this logic into a class `HttpsAgent` and have it calculate the Proxy or fallback to the default agent

**Comment by sashokbg** [comment]
Hello and thank you for your feedback - I have updated the PR accordingly :)

**Comment by sashokbg** [comment]
Are the failing tests related to this PR ?

**Comment by Meir017** [comment]
@sashokbg I don't think so. let me try and run this again

**Pull request merged**